lib/server.js

```diff
@@ -175,68 +176,24 @@ server.onPhantomPageCreate = function(req, res) {
     });
 };
 
-//We want to abort the request if it's a call to Google Analytics or other tracking services.
 /*
- * CODE DUPLICATION ALERT
- * Anything added to this if block has to be added to the if block
- * in server.onResourceRequestedCallback.
- * This if statment cannot be broken out into a helper method because
- * this method is serialized across the network to PhantomJS :(
- * Also, PhantomJS doesn't call onResourceError for an aborted request
+ * Note: PhantomJS doesn't call onResourceError for an aborted request
  */
-server.onResourceRequested = function (requestData, request) {
-    if ((/google-analytics.com/gi).test(requestData.url) ||
-        (/api.mixpanel.com/gi).test(requestData.url) ||
-        (/fonts.googleapis.com/gi).test(requestData.url) ||
-        (/stats.g.doubleclick.net/gi).test(requestData.url) ||
-        (/mc.yandex.ru/gi).test(requestData.url) ||
-        (/use.typekit.net/gi).test(requestData.url) ||
-        (/beacon.tapfiliate.com/gi).test(requestData.url) ||
-        (/js-agent.newrelic.com/gi).test(requestData.url) ||
-        (/api.segment.io/gi).test(requestData.url) ||
-        (/woopra.com/gi).test(requestData.url) ||
-        (/.ttf/gi).test(requestData.url) ||
-        (/static.olark.com/gi).test(requestData.url) ||
-        (/static.getclicky.com/gi).test(requestData.url) ||
-        (/fast.fonts.com/gi).test(requestData.url) ||
-        (/youtube.com\/embed/gi).test(requestData.url) ||
-        (/cdn.heapanalytics.com/gi).test(requestData.url) ||
-        (/googleads.g.doubleclick.net/gi).test(requestData.url) ||
-        (/pagead2.googlesyndication.com/gi).test(requestData.url)) {
-
-        request.abort();
+server.onResourceRequested = function (requestData, request, blockedResources) {
+    for(var i = 0,l = blockedResources.length; i < l; i++) {
+        var regex = new RegExp(blockedResources[i], 'gi');
+        if(regex.text(requestData.url)) {
+            request.abort();
+            requestData.aborted = true;
```

Comment:
This requestData is a shared object between `onResourceRequested` and `onResourceRequestedCallback`